File: invesdwin-context-parent/invesdwin-context-javascript/src/test/java/de/invesdwin/context/javascript/tests/callback/ParametersAndReturnsTestLong.js
print("getLong")
if(typeof getLong !== 'undefined')
	throw "getLong already defined!"
getLong = callback("getLong")
getLongType = typeof getLong
print(getLongType)
print(getLong)
if(getLongType !== 'number')
	throw "getLong not number!"
callback("setLong",getLong)

print("getLongVector")
if(typeof getLongVector !== 'undefined')
	throw "getLongVector already defined!"
getLongVector = callback("getLongVector")
getLongVectorType = typeof getLongVector[0]
print(getLongVectorType)
print(getLongVector)
if(getLongVectorType !== 'number')
	throw "getLongVector not number!"
callback("setLongVector",getLongVector)

print("getLongVectorAsList")
if(typeof getLongVectorAsList !== 'undefined')
	throw "getLongVectorAsList already defined!"
getLongVectorAsList = callback("getLongVectorAsList")
getLongVectorAsListType = typeof getLongVectorAsList[0]
print(getLongVectorAsListType)
print(getLongVectorAsList)
if(getLongVectorAsListType !== 'number')
	throw "getLongVectorAsList not number!"
callback("setLongVectorAsList",getLongVectorAsList)

print("getLongMatrix")
if(typeof getLongMatrix !== 'undefined')
	throw "getLongMatrix already defined!"
getLongMatrix = callback("getLongMatrix")
getLongMatrixType = typeof getLongMatrix[0][0]
print(getLongMatrixType)
print(getLongMatrix)
if(getLongMatrixType !== 'number')
	throw "getLongMatrix not number!"
callback("setLongMatrix",getLongMatrix)

print("getLongMatrixAsList")
if(typeof getLongMatrixAsList !== 'undefined')
	throw "getLongMatrixAsList already defined!"
getLongMatrixAsList = callback("getLongMatrixAsList")
getLongMatrixAsListType = typeof getLongMatrixAsList[0][0]
print(getLongMatrixAsListType)
print(getLongMatrixAsList)
if(getLongMatrixAsListType !== 'number')
	throw "getLongMatrixAsList not number!"
callback("setLongMatrixAsList",getLongMatrixAsList)

File: invesdwin-context-parent/invesdwin-context-jruby/src/test/java/de/invesdwin/context/jruby/tests/callback/ParametersAndReturnsTestLong.rb
puts("getLong")
unless (defined?(getLong)).nil?
	raise "getLong already defined!"
end
getLong = callback("getLong")
getLongType = getLong.class
puts(getLongType)
puts(getLong)
if(getLongType != Integer)
	raise "getLong not Integer!"
end
callback("setLong", getLong)

puts("getLongVector")
unless (defined?(getLongVector)).nil?
	raise "getLongVector already defined!"
end
getLongVector = callback("getLongVector")
getLongVectorType = getLongVector[0].class
puts(getLongVectorType)
puts(getLongVector)
if(getLongVectorType != Integer)
	raise "getLongVector not Integer!"
end
callback("setLongVector", getLongVector)

puts("getLongVectorAsList")
unless (defined?(getLongVectorAsList)).nil?
	raise "getLongVectorAsList already defined!"
end
getLongVectorAsList = callback("getLongVectorAsList")
getLongVectorAsListType = getLongVectorAsList[0].class
puts(getLongVectorAsListType)
puts(getLongVectorAsList)
if(getLongVectorAsListType != Integer)
	raise "getLongVectorAsList not Integer!"
end
callback("setLongVectorAsList", getLongVectorAsList)

puts("getLongMatrix")
unless (defined?(getLongMatrix)).nil?
	raise "getLongMatrix already defined!"
end
getLongMatrix = callback("getLongMatrix")
getLongMatrixType = getLongMatrix[0][0].class
puts(getLongMatrixType)
puts(getLongMatrix)
if(getLongMatrixType != Integer)
	raise "getLongMatrix not Integer!"
end
callback("setLongMatrix", getLongMatrix)

puts("getLongMatrixAsList")
unless (defined?(getLongMatrixAsList)).nil?
	raise "getLongMatrixAsList already defined!"
end
getLongMatrixAsList = callback("getLongMatrixAsList")
getLongMatrixAsListType = getLongMatrixAsList[0][0].class
puts(getLongMatrixAsListType)
puts(getLongMatrixAsList)
if(getLongMatrixAsListType != Integer)
	raise "getLongMatrixAsList not Integer!"
end
callback("setLongMatrixAsList", getLongMatrixAsList)
